### ISSUE TITLE
Problem with equal-hash? -- Segmentation fault: 11

### ISSUE THREAD

**Issue by ktakashi** [ktakashi/sagittarius-scheme#187]
**[Original report](https://bitbucket.org/ktakashi/sagittarius-scheme/issues/182) by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Reported on Github.

https://github.com/ktakashi/sagittarius-scheme/issues/1

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

`eqv?` of record causes stack overflow. The following is a simple example:
```
#!scheme
(import (rnrs))
(define-record-type (<pare> kons pare?)
  (fields (mutable a kar set-kar!)
	  (mutable d kdr set-kdr!)))

(let ((a (kons 'a 'b))
      (b (kons 'a 'b)))
  (set-kdr! a a)
  (set-kdr! b b)
  (eqv? a b))
;; -> SEGV
```

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Both R6RS and R7RS say that comparison of record by `eqv?` returns #t only if the memory location is the same. So basically, the error case should always return #f.

NB: this also means `equal?` must also return #f for the sake of R6RS. (or more precisely, the R6RS test suites.)

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

eqv? shouldn't inspect record fields. (Fixes #182)
Comforming R6RS equal? strictly (breaking backward compatibility)
